source/agora/consensus/Validation.d

```diff
@@ -33,23 +33,24 @@ public alias UTXOFinder = scope const(Output)* delegate (Hash hash, size_t index
         findUTXO = delegate for finding `Output`
 
     Return:
-        Return true if this transaction is verified.
+        `null` if the transaction is valid, a string explaining the reason it
+        is invalid otherwise.
 
 *******************************************************************************/
 
-public bool isValid (const Transaction tx, UTXOFinder findUTXO)
+public string invalidReason (const Transaction tx, UTXOFinder findUTXO)
```

Comment:
I dislike the name of this function. It seems to imply (on first glance) that the transaction/block is already invalid. And it's also not a verb, and hard to memorize. `validate` or `checkValid` or something like that might be better, but I can't suggest a really good name for this..